lms/djangoapps/completion/utils.py

```diff
@@ -57,7 +59,17 @@ def get_contact_id(user):
         activities = len(required_ids)
 
         for idx, item in enumerate(required_ids):
-            fieldnames.append("Required Activity {}".format(idx + 1))
+            block_type, block_id = item.rsplit("+block@")
+            locator = BlockUsageLocator(self.course_key, block_type, block_id)
+
+            try:
+                block = modulestore().get_item(locator)
+            except ItemNotFoundError as item_error:
+                logger.warn("The provider id is not valid, error %s", item_error)
+                continue
+
+            block_name = block.display_name
```

Comment:
You can merge this assignation on the next line, since the variable "block_name" it's only used there. @andrey-canon